src/api/routers/generation.py

```
@@ -0,0 +1,714 @@
+                "model_status": {
+                    "loaded": model_loaded,
+                    "model_path": model_info.get("model_path", ""),
+                    "load_time": model_info.get("load_time", 0),
+                    "context_length": model_info.get("context_length", 0),
+                    "quantization": model_info.get("quantization", "unknown"),
+                    "threads": model_info.get("threads", 0)
+                },
+                "generation_stats": generation_stats,
+                "system_stats": {
+                    "uptime_seconds": uptime,
+                    "memory_usage_mb": self._get_memory_usage(),
```

Ruff (0.13.1)

Undefined name `self`

(F821)